### PR TITLE
Only test supported versions of Ruby on Travis

### PR DESCRIPTION
**Pull request by bdewater**

### CODE REVIEW

**Comment by bdewater** [comment]
Some bundling errors, Ruby 2.4 needs json >= 2.0 because of the Fixnum and BigNum unification but it tries to install 1.8.3 for some reason. 4.0 and 5.0 (strangely enough) are fine in this regard but produce an actual failure that is related to https://github.com/ruby/ruby/commit/ce635262f53b760284d56bb1027baebaaec175d1

**Comment by bdewater** [comment]
Waiting for https://github.com/rails/rails/commit/47c4924 to be released in Rails 4.2.8 (should be soon), CI against the 4-2-stable branch is green.

File: lib/offsite_payments/integrations/sage_pay_form.rb [hunk 1]
```diff
@@ -160,8 +160,8 @@ def create_crypt_field(fields, key)
           parts = fields.map { |k, v| "#{k}=#{sanitize(k, v)}" unless v.nil? }.compact.shuffle
           parts.unshift(sage_encrypt_salt(key.length, key.length * 2))
           sage_encrypt(parts.join('&'), key)
-        rescue OpenSSL::Cipher::CipherError => e
-          if e.message == 'key length too short'
+        rescue OpenSSL::Cipher::CipherError, ArgumentError => e
+          if e.message == 'key length too short' || e.message == 'key must be 16 bytes'
```
Comment (bdewater) [inline lib/offsite_payments/integrations/sage_pay_form.rb]: Ruby 2.4 will raise a different error then older Rubies, this way we keep the same behaviour.

**Comment by pi3r** [comment]
do you know why ruby 2.4 can't install the correct version of json?

**Comment by bdewater** [comment]
JSON 2.0 has the required changes for Ruby 2.4's unification of Fixnum and Bignum into Integer. Rails 4.2.7.1 requires ~> 1.7, which leads to Bundler installing 1.8.3 and that blows up. Rails 4.2.8 will remove the JSON gem requirement.

**Pull request merged**